src/components/Drawer/__test__/drawer.spec.js

```diff
@@ -0,0 +1,262 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import CounterManager from '../../../libs/counterManager';
+import {
+    disableBodyScroll,
+    enableBodyScroll,
+    clearAllBodyScrollLocks,
+} from '../../../libs/scrollController';
+import Drawer from '../';
+import StyledContent from '../styled/content';
+import StyledCloseButton from '../styled/closeButton';
+
+const DrawerState = {
+    OPENING: 0,
+    OPENED: 1,
+    CLOSING: 2,
+    CLOSED: 3,
+};
+
+jest.mock('../../../libs/counterManager', () => ({
+    increment: jest.fn(),
+    decrement: jest.fn(),
+    hasModalsOpen: jest.fn(() => false),
+}));
+
+jest.mock('../../../libs/scrollController', () => ({
+    disableBodyScroll: jest.fn(),
+    enableBodyScroll: jest.fn(),
+    clearAllBodyScrollLocks: jest.fn(),
+}));
+
+describe('<Drawer />', () => {
+    const setState = jest.fn();
+    const useStateSpy = jest.spyOn(React, 'useState');
+    useStateSpy.mockImplementation(init => [init, setState]);
+
+    beforeEach(() => {
+        disableBodyScroll.mockReset();
+        enableBodyScroll.mockReset();
+        clearAllBodyScrollLocks.mockReset();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render the children passed', () => {
+        const component = mount(
+            <Drawer isOpen>
+                <p data-id="drawer-children">drawer content</p>
+            </Drawer>,
+        );
+        setTimeout(() => {
+            expect(component.find('p[data-id="drawer-children"]').text()).toBe('drawer content');
```

Comment:
The use of `expect.assertions(1);` is important because ensure the assertions always runs

Reply:
so it should be include in all tests?

Reply:
it is only needed for async testing:
https://jestjs.io/docs/en/asynchronous